examples/FMP_simple_temp.jl

```diff
@@ -0,0 +1,252 @@
+# - :O = object (has no velocity, cannot move, agents must go around)
+# - :T = target (no velocity, just a projection of agent target zone)
+
+using Agents, Random, Plots, LinearAlgebra, Colors, ProgressMeter
+
+ENV["GKSwstype"]="nul"
+gr()
+cd(@__DIR__)
+
+mutable struct FMP_Agent <: AbstractAgent
+    id::Int
+    pos::NTuple{2, Float64}
+    vel::NTuple{2, Float64}
+    tau::NTuple{2, Float64}
+    color::String
+    type::Symbol
+    radius::Float64
+    SSdims::NTuple{2, Float64}  ## include this for plotting
+    Ni::Array{Tuple{Int64, NTuple{2, Float64}}} ## array of tuples with agent ids and agent positions
+end
+
+# ## Defining the Model
+
+# Next, we define the model. The FMP algorithm has several parameters defined
+# in the original paper which are stored in a struct. Typical values can be
+# generated using the `FMP_Parameter_Init` function and loaded into the model
+# properties. 
+
+# Additionally, note that we pass in `FMP_Update_Vel` as the `update_vel!`
+# keyword argument when we initialize our `ContinuousSpace`.
+
+## define AgentBasedModel (ABM)
+
+function FMP_Model()
+    properties = Dict(:FMP_params=>FMP_Parameter_Init(),
+                      :dt => 0.01,
+                      :num_agents=>100,
+                      :num_steps=>1500,
+                      :step_inc=>2,
+                     )
+
+    #space2d = ContinuousSpace((1,1); periodic = true, update_vel! = FMP_Update_Vel)  #BONE commented this out
+    space2d = ContinuousSpace((1,1); periodic = true)
+    model = ABM(FMP_Agent, space2d, properties=properties)
+    return model
+end
+
+# ## Adding Agents to the Model
+
+# Now that we have defined a model, lets add some agents into it. In the
+# following example, we add agents around the perimeter of a circle. We add
+# their target position (tau) across the circle from each agent. Note that the
+# arguments passed to `add_agent!` correspond to the `AbstractAgent` struct
+# we defined earlier. Note that `:A` (agent types) experience navigational
+# forces and repulsive forces. `:T` (target types) do not experience any
+# forces.
+
+# Also note that the `color` parameter doesn't do anything unless we use some
+# plotting functionalities (described later in this tutorial)
+
+## add agents to model
+## determine circle params
+# ## Plotting Utilities
+
+# The above animation isn't very pretty looking - luckily we can pass in some
+# plotting parameters. This section defines some plotting utilities that help
+# with make the generated plot look a little more convincing. 
+
+# First, lets define a function that assigns color based on agent type.
+
+using Colors
+"""
+This function is a utility function that takes an agent and overwrites its color if it is of type :T (target) and gives it a grey color for display purposes.
+"""
+function PlotABM_ColorUtil(a::AbstractAgent)
+    if a.type == :A || a.type == :O
+        return a.color
+    else
+        return "#ffffff"
+    end
+end
```

Comment:
all of this function can be made as `f(a) = a.type in (:A, :B) ? a.color : "#ffffff"`